File: app/src/main/java/io/mgba/Activities/SplashActivity.java
package io.mgba.Activities;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import io.mgba.Model.System.PreferencesManager;
import io.mgba.mgba;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        boolean hasDoneSetup = ((mgba)getApplication()).getPreference(PreferencesManager.SETUP_DONE, false);

        Intent it = new Intent(this, hasDoneSetup ? MainActivity.class : IntroActivity.class);
        startActivity(it);
        finish();
    }

}

File: app/src/main/java/io/mgba/ui/activities/SplashActivity.kt
package io.mgba.ui.activities

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import io.mgba.model.system.PreferencesManager

class SplashActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val hasDoneSetup = PreferencesManager[PreferencesManager.SETUP_DONE, false]

        val it = Intent(this, if (hasDoneSetup) MainActivity::class.java else IntroActivity::class.java)
        startActivity(it)
        finish()
    }

}
